src/components/app/accountsState.ts

```diff
@@ -0,0 +1,80 @@
+// eslint-disable-next-line max-classes-per-file
+import { BudgetTrackingEvent } from '@/originalBudgetTrackingApp';
+import { Levels } from '@/components/shared/log/types';
+import { UnwrapRef } from '@vue/composition-api';
+import { AccountStatus, AccountType } from '@/originalBudgetTrackingApp/eventEmitters/EventEmitter';
+
+export class AccountState {
+  id: string;
+
+  name: string;
+
+  status: AccountStatus;
+
+  events: BudgetTrackingEvent[];
+
+  constructor({ id, name }) {
+    this.id = id;
+    this.name = name;
+    this.status = AccountStatus.IDLE;
+    this.events = [];
+  }
+
+  clear() {
+    this.status = AccountStatus.IDLE;
+    this.events = [];
+  }
+}
+
+export class ImporterState extends AccountState {
+
+}
+
+export class ExporterState extends AccountState {
+
+}
+
+export class AccountsState {
```

Comment:
Maybe rename to `ScrapingState` to different from the very similar name `AccountState`?

Reply:
I thought about it. But it isn't scraping state because it includes the exporters too.

It actually is AccountsState because AccountState is of a single account and this is basically a container for many of those AccountState objects :) 

WDYT?

Reply:
Fine, it's not important.

Just because I didn't understand it immediately until I figure that there is class `AccountsState` and `AccountState`.

In VSCode you don't have problems like that.